h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -23,10 +24,143 @@ class SearchBarController extends Controller {
+        try{
+          tagSuggestions = JSON.parse(tagSuggestionJSON.innerHTML.trim());
+        }catch(e){
+          console.error('Could not parse .js-tag-suggestions JSON content', e);
+        }
+      }
+
+      let tagsList = ((tagSuggestions) || []).map((item)=>{
+        return Object.assign(item, {
+          type: TAG_TYPE,
+          title: item.tag, // make safe
+          matchOn: stringUtil.fold(stringUtil.normalize(item.tag)),
+          usageCount: item.count || 0,
+        });
+      });
+
+      // groupSuggestions are made available by the scoped template data.
+      // see search.html.jinja2 for definition
+      const groupSuggestionJSON = document.querySelector('.js-group-suggestions');
+      let groupSuggestions = [];
+
+      if(groupSuggestionJSON){
+        try{
+          groupSuggestions = JSON.parse(groupSuggestionJSON.innerHTML.trim());
+        }catch(e){
+          console.error('Could not parse .js-group-suggestions JSON content', e);
+        }
+      }
+
+      let groupsList = ((groupSuggestions) || []).map((item)=>{
+        return Object.assign(item, {
+          type: GROUP_TYPE,
+          title: item.name, // make safe
+          matchOn: stringUtil.fold(stringUtil.normalize(item.name)),
+          pubid: item.pubid,
+          name: item.name,
+        });
+      });
+
+      return explanationList.concat(tagsList, groupsList);
+    })();
+
     var getTrimmedInputValue = () => {
       return this._input.value.trim();
     };
 
+
+    /**
+     * given a lozenge value for a group: find the pubid, if we can
+     *  for that value. If we find this is a pub value or we do not
+     *  find a match with the list we have, we will give back the
+     *  full lozenge value
+     */
+    var getInputAndDisplayValsForGroup = (groupLoz) => {
+      let groupVal = groupLoz.substr(groupLoz.indexOf(':') + 1).trim();
+      let inputVal = groupVal.trim();
+      let displayVal = groupVal;
+      let wrapQuotesIfNeeded = function(str){
+        return str.indexOf(' ') > -1 ? `"${str}"` : str;
+      };
+
+
+      // remove quotes from value
+      if(groupVal[0] === '"' || groupVal[0] === '\''){
+        groupVal = groupVal.substr(1);
+      }
+      if(groupVal[groupVal.length - 1] === '"' || groupVal[groupVal.length - 1] === '\''){
+        groupVal = groupVal.slice(0, -1);
+      }
+
+      const matchVal = stringUtil.fold(stringUtil.normalize(groupVal)).toLowerCase();
+
+      // NOTE: We are pushing a pubid to lowercase here. These ids are created by us
+      // in a random generation case-sensistive style. Theoretically, that means
+      // casting to lower could cause overlaps of values like 'Abc' and 'aBC' - making
+      // them equal to us. Since that is very unlikely to occur for one user's group
+      // set, the convenience of being defensive about bad input/urls is more valuable
+      // than the risk of overlap.
+      const matchByPubid = this._suggestionsMap.find((item)=>{
```

Comment:
that is correct it's needed to resolve existing group query lozenge values - which will be pubid more often than not.